### PR TITLE
Fixed wrong recursive call in _substmerge()

### PR DESCRIPTION
**Pull request by alex-ber**
Sorry, in my https://github.com/zerwes/hiyapyco/pull/37 there was bug, I've called wrong method. This is the fix.

### CODE REVIEW

File: hiyapyco/__init__.py [hunk 1]
```diff
@@ -242,10 +242,8 @@ def _load_data(self, _usedefaultyamlloader, f):
                     self._data = self._simplemerge(self._data, ydata)
                 elif self.method == METHOD_MERGE:
                     self._data = self._deepmerge(self._data, ydata)
-                elif self.method == METHOD_SUBSTITUTE:
-                    self._data = self._substmerge(self._data, ydata)
                 else:
-                    raise HiYaPyCoInvocationException('unknown merge method \'%s\'' % self.method)
```
Comment (alex-ber) [inline hiyapyco/__init__.py]: Take a closer look on this, I didn't touch this line.

File: hiyapyco/__init__.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ class HiYaPyCoImplementationException(Exception):
 # you may set this to something suitable for you
 jinja2env = Environment(undefined=Undefined)
 
-METHODS = { 'METHOD_SIMPLE':0x0001, 'METHOD_MERGE':0x0002, 'METHOD_SUBSTITUTE':0x0003, }
```
Comment (alex-ber) [inline hiyapyco/__init__.py]: I didn't touch this either, but this is minor change.

Comment (alex-ber) [reply hiyapyco/__init__.py]: I've incorporated your fix in my latest commit.

File: hiyapyco/__init__.py (hunk 1 above)
Comment (alex-ber) [reply hiyapyco/__init__.py]: I've incorporated your fix in my latest commit.

**Comment by lucas-bremond** [comment]
Could the maintainer merge this one please?

**Comment by alex-ber** [comment]
@lucas-bremond, meanwhile, you can use my "sort-of-fork" here https://github.com/alex-ber/AlexBerUtils/blob/master/alexber/utils/ymlparsers.py Documentation is available here https://medium.com/analytics-vidhya/my-ymlparsers-module-88221edf16a6

**Comment by alex-ber** [comment]
It is almost a year was passed and my Pull Request is still not merged. I'm achieving my project.

**Comment by zerwes** [comment]
merged and resolved conflicts on current master in branch [pr38](https://github.com/zerwes/hiyapyco/tree/pr38)
:-1: failing tests result (https://github.com/zerwes/hiyapyco/actions/runs/2531814314)

**Pull request closed without merging**